src/wasi_libc.zig

```diff
@@ -51,16 +51,20 @@ pub fn buildWasiLibcSysroot(comp: *Compilation) !void {
             }),
         });
 
-        var comp_sources = std.ArrayList(Compilation.CSourceFile).init(arena);
         for (crt_src_files) |file_path| {
+            var comp_sources = std.ArrayList(Compilation.CSourceFile).init(arena);
             try comp_sources.append(.{
                 .src_path = try comp.zig_lib_directory.join(arena, &[_][]const u8{
                     "libc", try sanitize(arena, file_path),
                 }),
                 .extra_flags = args.items,
             });
+            try comp.build_crt_file(
+                file_path[std.mem.lastIndexOf(u8, file_path, "/").? + 1 .. std.mem.lastIndexOf(u8, file_path, ".").?],
+                .Obj,
+                comp_sources.items,
+            );
```

Comment:
These changes are no longer necessary and are fixed in https://github.com/ziglang/zig/pull/8992

Reply:
sounds good! so should I wait until that PR is merged?

Reply:
Actually, feel free to rebase onto my branch, and we can carry on from there. Feel free to ping me on Discord if you have any questions about the changes in that PR or whatnot!